src/ec/packages/ec-component-timeline/timeline.html.twig

```diff
@@ -0,0 +1,120 @@
+{% spaceless %}
+
+{#
+Parameters:
+  - "toggleCollapsed": (string) (default: 'Show %d more items')
+  - "toggleExpanded": (string) (default: 'Show %d items')
+  - "hide" (array): (default: undefined)
+    - "from": (integer) (default: undefined) Positive integer
+    - "to": (integer) (default: undefined) Negative integer
+  - "items" (array) (default: []):
+    - "id": (string) (default: '')
+    - "label": (string) (default: '')
+    - "title": (string) (default: '')
+    - "content": (block) (default: '')
+  - "icon_path" (string) (default: ''): file containing the svg icons
+  - "extra_classes" (string) (default: '')
+  - "extra_attributes" (array) (default: []): format:
+    - "name" (string) (default: '')
+    - "value" (string) (default: '')
+#}
+
+{# Internal properties #}
+
+{% set _css_class = 'ecl-timeline2' %}
+{% set _extra_attributes = '' %}
+{% set _items = items|default([]) %}
+{% set _icon_path = icon_path|default('') %}
+{% set _toggle_collapsed = toggleCollapsed|default('Show %d more items') %}
+{% set _toggle_expanded = toggleExpanded|default('Hide %d items') %}
+{% set _hide_from = hide.from|default(items|length) %}
+{% set _hide_to = hide.to|default(items|length) %}
+
+{% set display_amount = items|length %}
+{% set hide_amount = 0 %}
+{% set bottom_amount = 0 %}
+
+{% if _hide_from > 0 %}
+  {% set bottom_amount = _hide_to|abs %}
```

Comment:
what happens when `hide.to` is positive?

let's say I have 15 items, I want to hide from the 3rd to the 13th, I provide `hide: { from 3, to 13 }`
=> `bottom_amount = 13`
=> `countbase = 15 - 13 = 2`
=> `display_amount` = 3`
=> `hide_amount`= 2 - 3 = -1`
and nothing is hidden

the logic could be simplified

```
let first_item_to_hide, last_item_to_hide
if hide.from and hide.to are defined
  if hide.from < 0
    first_item_to_hide = items|length + hide.from
  else 
    first_item_to_hide = hide.from
  endif
  if hide.to < 0
    last_item_to_hide = items|length + hide.to
  else 
    last_item_to_hide = hide.to
  endif
endif

# now we have the right indices, let's make sure there's something to hide
bool hidden_items = first_item_to_hide < last_item_to_hide

# let's iterate
for item in items
  let item_css_class 
  if hidden_items == true
    if loop.index == last_item_to_hide + 1
      # print button
    else if first_item_to_hide < loop.index and loop.index < last_item_to_hide
      item_css_class = ecl-timeline2__item--collapsed
    endif
  endif
  # print item with item_css_class
endfor
```

I didn't care about the exact indices, but the overall logic could be simplified like this couldn't it?

Reply:
I have updated the calculation to take care of both scenarios.